pyproject.toml

```diff
@@ -53,6 +53,7 @@ dependencies = [
     "importlib_metadata>=3.6",
     "packaging",
     "sopel-help>=0.4.0",
+    "sopel-iplookup",
```

Comment:
Generally speaking, if we're pulling a plugin out of core, it doesn't become a requirement. The `help` plugin has been the only exception to that rule, because it's actually pretty important for users to be able to get command lists & usage information out of the box.

For cases like this where we're extracting the plugin due to a dependency problem, we _really_ shouldn't make the new external package a requirement. The new plugin package will still depend on geoip2, which depends on aiohttp, which doesn't yet install cleanly on Python 3.12 and would still leave #2516 blocked after this change.